services/galley/src/Galley/Run.hs

```diff
@@ -0,0 +1,55 @@
+module Galley.Run (run) where
+
+import           Imports
+
+import           Cassandra                          (runClient, shutdown)
+import           Cassandra.Schema                   (versionCheck)
+import           Control.Exception                  (finally)
+import           Control.Lens                       ((^.))
+import           Data.Metrics.Middleware.Prometheus (waiPrometheusMiddleware)
+import           Data.Metrics.WaiRoute              (treeToPaths)
+import           Data.Misc                          (portNumber)
+import           Data.Text                          (unpack)
+import           Network.Wai                        (Middleware)
+import           Network.Wai.Utilities.Server
+import           Util.Options
+import qualified Control.Concurrent.Async           as Async
+import qualified Data.Metrics.Middleware            as M
+import qualified Network.Wai.Middleware.Gunzip      as GZip
+import qualified Network.Wai.Middleware.Gzip        as GZip
+import qualified System.Logger                      as Log
+
+import           Galley.API          (sitemap)
+import qualified Galley.API.Internal as Internal
+import qualified Galley.App          as App
+import Galley.App
+import qualified Galley.Data         as Data
+import           Galley.Options      (Opts, optGalley)
+
+run :: Opts -> IO ()
+run o = do
+    m <- M.metrics
+    e <- App.createEnv m o
+    let l = e ^. App.applog
+    s <- newSettings $ defaultServer (unpack $ o ^. optGalley.epHost)
+                                     (portNumber $ fromIntegral $ o ^. optGalley . epPort)
+                                     l
+                                     m
+    runClient (e^.cstate) $
+        versionCheck Data.schemaVersion
+    d <- Async.async $ evalGalley e Internal.deleteLoop
+    let rtree    = compile sitemap
+        app r k  = runGalley e r (route rtree r k)
+        measured :: Middleware
+        measured = measureRequests m (treeToPaths rtree)
+        middlewares :: Middleware
+        middlewares = waiPrometheusMiddleware sitemap
```

Comment:
Anyone know what the ordering of the middleware should be?

Reply:
it is important to catch errors early enough.  the prometheus middleware should report metrics on crashes, so by the time it is called, catchErrors should be done with its thing.

Same goes for possibly one or two other middlewares, somewhere.  Not sure, just apply common sense?

Reply:
(why is there a zip followed by an unzip?  i presume it's been there before, but it looks odd.)

Reply:
@fisx `zip` compresses the response body; `unzip` uncompresses the request body 😄